src/domain/accounts/index.types.d.ts

```diff
@@ -0,0 +1,16 @@
+type AccountLevel =
+  typeof import("./index").AccountLevel[keyof typeof import("./index").AccountLevel]
```

Comment:
I believe we can type this directly here to get the constant values assigned to each field, although I'm not sure if we deliberately did it this way so that the type is defined in only one place.

If not, typescript should allow something like `type a = "Alice"` to be type string restricted to single string value `'Alice'`.

_(This comment would be for all places where we use this pattern)_

```suggestion
type AccountLevelType = {
  One: 1,
  Two: 2,
}
```

Reply:
What I want is to be able to reference the constants in code without typing the strings:
```
const level = AccountLevelType.One
```

if we put the definition in the `.types.d.ts` file we would have the definition in 2 places or have to hard code values everywhere:
```
const level = 1
```

enums have other drawbacks as they are not native to JS https://www.typescriptlang.org/docs/handbook/enums.html#objects-vs-enums.

All things considered I chose this pattern for collections of constants but perhaps we can discuss it in the team at daily.